src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -76,9 +87,78 @@ internal abstract class JsonPropertyInfo
         internal virtual void GetPolicies(JsonSerializerOptions options)
         {
             DetermineSerializationCapabilities(options);
+            DeterminePropertyName(options);
             IgnoreNullValues = options.IgnoreNullValues;
         }
 
+        private void DeterminePropertyName(JsonSerializerOptions options)
+        {
+            if (PropertyInfo != null)
+            {
+                JsonPropertyNameAttribute nameAttribute = GetAttribute<JsonPropertyNameAttribute>();
+                if (nameAttribute != null)
+                {
+                    NameAsString = nameAttribute.Name;
+                    
+                    // This is detected and thrown by caller.
+                    if (NameAsString == null)
+                    {
+                        return;
+                    }
+                }
+                else if (options.PropertyNamingPolicy != null)
+                {
+                    NameAsString = options.PropertyNamingPolicy.ConvertName(PropertyInfo.Name);
+
+                    // This is detected and thrown by caller.
+                    if (NameAsString == null)
```

Comment:
Should this a Debug.Assert then?

Reply:
The caller is the method calling this, not the end user. It is just exiting the method.